packages/expo-crypto/ios/EXCrypto/EXCrypto.m

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019-present 650 Industries. All rights reserved.
+
+#import <EXCrypto/EXCrypto.h>
+#import <CommonCrypto/CommonDigest.h>
+
+@implementation EXCrypto
+
+EX_EXPORT_MODULE(ExpoCrypto);
+
+EX_EXPORT_METHOD_AS(digestStringAsync,
+                    digestStringAsync:(NSString *)algorithm
+                    data:(NSString *)data
+                    options:(NSDictionary *)options
+                    resolver:(EXPromiseResolveBlock)resolve
+                    rejecter:(EXPromiseRejectBlock)reject)
+{
+  NSString *encoding = options[@"encoding"];
+
+  int digestLength = [EXCrypto digestLengthForJSAlgorithm:algorithm];
+  if (digestLength == -1) {
+    reject(@"ERR_CRYPTO_DIGEST", [NSString stringWithFormat:@"Invalid hashing algorithm: %@", algorithm], nil);
+    return;
+  }
+  
+  NSData *_data = [data dataUsingEncoding:NSUTF8StringEncoding];
+  uint8_t digest[digestLength];
+  
+  if (digestLength == CC_MD2_DIGEST_LENGTH) {
+    CC_MD2(_data.bytes, (CC_LONG)_data.length, digest);
+  } else if (digestLength == CC_MD4_DIGEST_LENGTH) {
+    CC_MD4(_data.bytes, (CC_LONG)_data.length, digest);
+  } else if (digestLength == CC_MD5_DIGEST_LENGTH) {
+    CC_MD5(_data.bytes, (CC_LONG)_data.length, digest);
+  } else if (digestLength == CC_SHA1_DIGEST_LENGTH) {
+    CC_SHA1(_data.bytes, (CC_LONG)_data.length, digest);
+  } else if (digestLength == CC_SHA224_DIGEST_LENGTH) {
+    CC_SHA224(_data.bytes, (CC_LONG)_data.length, digest);
+  } else if (digestLength == CC_SHA256_DIGEST_LENGTH) {
+    CC_SHA256(_data.bytes, (CC_LONG)_data.length, digest);
+  } else if (digestLength == CC_SHA384_DIGEST_LENGTH) {
+    CC_SHA384(_data.bytes, (CC_LONG)_data.length, digest);
+  } else if (digestLength == CC_SHA512_DIGEST_LENGTH) {
+    CC_SHA512(_data.bytes, (CC_LONG)_data.length, digest);
+  }
+  
+  if ([encoding isEqualToString:@"hex"]) {
+    NSMutableString *output = [NSMutableString stringWithCapacity:digestLength * 2];
+    for (int i = 0; i < digestLength; i++) {
+      [output appendFormat:@"%02x", digest[i]];
+    }
+    resolve(output);
+    return;
+  } else if ([encoding isEqualToString:@"base64"]) {
+    NSData *originalData = [NSData dataWithBytes:digest length:digestLength];
+    NSString *output = [originalData base64EncodedStringWithOptions:NSDataBase64EncodingEndLineWithLineFeed];
```

Comment:
```suggestion
    NSString *output = [originalData base64EncodedStringWithOptions:0];
```

Don't need any options since we don't want to wrap lines. Also verify this doesn't have a trailing newline, and trim if it does.